src/service.ts

```diff
@@ -6,6 +6,42 @@ import * as k8s from './imports/k8s';
 import * as ingress from './ingress';
 import * as pod from './pod';
 
+/**
+ * Interface for Service constructs.
+ * This interface allows users to interact with a Service without having to create one directly.
+ */
+export interface IService extends base.IResource {
+  /**
+   * The ports this service binds to.
+   */
+  readonly ports: ServicePort[];
+
+  /**
+   * The first port of the service, if available.
+   */
+  readonly port?: number;
```

Comment:
Do we really need this on the interface? From what I could tell, just `ports` is needed. Actually the fact both `port` and `ports` even exist is for backward compat; but the new interface doesn't have to suffer from this.